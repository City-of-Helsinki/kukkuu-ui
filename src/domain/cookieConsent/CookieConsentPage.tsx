import { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';

import PageWrapper from '../app/layout/PageWrapper';
import CookieConsent from './CookieConsent';

const CookieConsentPage: FunctionComponent = () => {
  const { t } = useTranslation();

  return (
    <PageWrapper>
      <CookieConsent isModal={false} appName={t('appName')} />
    </PageWrapper>
  );
};

export default CookieConsentPage;
