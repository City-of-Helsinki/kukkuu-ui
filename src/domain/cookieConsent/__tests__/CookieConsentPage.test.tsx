import { render } from '../../../common/test/testingLibraryUtils';
import CookieConsentPage from '../CookieConsentPage';

it('renders snapshot correctly', () => {
  const { container } = render(<CookieConsentPage />);
  expect(container).toMatchSnapshot();
});
