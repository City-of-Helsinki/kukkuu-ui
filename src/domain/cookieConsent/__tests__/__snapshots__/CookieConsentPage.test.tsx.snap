// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders snapshot correctly 1`] = `
<div>
  <div
    class="_pageWrapper_e42bcd"
  >
    <div
      class="_container_533ff0"
    >
      <div
        class="CookieConsent-module_page__1BCN2"
        data-testid="cookie-consent"
      >
        <div
          class="CookieConsent-module_content__3QPI4"
          id="cookie-consent-content"
        >
          <div
            class="CookieConsent-module_mainContent__35KR_"
            data-testid="cookie-consent-information"
          >
            <h1
              class="heading-xl"
            >
              Kulttuurin kummilapset käyttää evästeitä
            </h1>
            <p>
              Tämä sivusto käyttää välttämättömiä evästeitä sivun perustoimintojen ja suorituskyvyn varmistamiseksi. Lisäksi käytämme kohdennusevästeitä käyttäjäkokemuksen parantamiseksi, analytiikkaan ja yksilöidyn sisällön näyttämiseen.
            </p>
          </div>
          <div
            class="CookieConsent-module_textContent__1H03u"
            data-testid="cookie-consent-details"
          >
            <span
              aria-level="2"
              class="CookieConsent-module_emulatedH2__3BlmG helpers_heading-m__9vrUW"
              role="heading"
            >
              Tietoa sivustolla käytetyistä evästeistä
            </span>
            <p>
              Sivustolla käytetyt evästeet on luokiteltu käyttötarkoituksen mukaan. Alla voit lukea eri luokista ja sallia tai kieltää evästeiden käytön.
            </p>
            <div
              class="CookieConsent-module_consentCategory__iJprz"
            >
              <div
                class="CookieConsent-module_titleWithCheckbox__1QVs0"
              >
                <div
                  class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz"
                  style="--label-font-size: var(--fontsize-heading-m);"
                >
                  <input
                    aria-describedby="required-consents-description"
                    checked=""
                    class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
                    data-testid="required-consents-checkbox"
                    disabled=""
                    id="required-consents-checkbox"
                    name="required-consents-checkbox"
                    type="checkbox"
                    value=""
                  />
                  <label
                    class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
                    for="required-consents-checkbox"
                  >
                    Välttämättömät evästeet
                  </label>
                </div>
              </div>
              <p
                aria-hidden="true"
              >
                Välttämättömien evästeiden käyttöä ei voi kieltää. Ne mahdollistavat sivuston toiminnan ja vaikuttavat käytettävyyteen.
              </p>
              <div
                id="required-consents-description"
                style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
              >
                Välttämättömien evästeiden käyttöä ei voi kieltää. Ne mahdollistavat sivuston toiminnan ja vaikuttavat käytettävyyteen.
              </div>
              <ul
                class="CookieConsent-module_list__145_H"
              >
                <li>
                  <div
                    class="CookieConsent-module_consentGroup__2i8Av"
                  >
                    <div
                      class="CookieConsent-module_titleWithCheckbox__1QVs0"
                    >
                      <div
                        class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz"
                        style="--label-font-size: var(--fontsize-heading-s);"
                      >
                        <input
                          aria-describedby="required-consents-group-0-description"
                          checked=""
                          class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
                          data-testid="required-consents-group-0-checkbox"
                          disabled=""
                          id="required-consents-group-0-checkbox"
                          name="required-consents-group-0-checkbox"
                          type="checkbox"
                          value=""
                        />
                        <label
                          class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
                          for="required-consents-group-0-checkbox"
                        >
                          Perustoimintoihin liittyvät evästeet
                        </label>
                      </div>
                    </div>
                    <div
                      class="CookieConsent-module_consentGroupContent__1x4Uq"
                    >
                      <p
                        aria-hidden="true"
                      >
                        Perustoimintoihin liittyviä evästeitä ei voi hylätä. Ne mahdollistavat sivuston kunnollisen toimivuuden ja vaikuttavat käytettävyyteen.
                      </p>
                      <div
                        id="required-consents-group-0-description"
                        style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
                      >
                        Perustoimintoihin liittyviä evästeitä ei voi hylätä. Ne mahdollistavat sivuston kunnollisen toimivuuden ja vaikuttavat käytettävyyteen.
                      </div>
                      <button
                        aria-expanded="false"
                        class="CookieConsent-module_accordionButton__3R8Nn"
                        data-testid="required-consents-group-0-details-toggler"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          aria-label="angle-down"
                          class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                          role="img"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M12 13.5L17 8.5L18.5 10L12 16.5L5.5 10L7 8.5L12 13.5Z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </button>
                      <div
                        aria-label="Perustoimintoihin liittyvät evästeet"
                        class="Card-module_card__1WbKx custom-theme-1"
                        role="region"
                        style="display: none;"
                      >
                        <div
                          class="CookieConsent-module_dataTableContainer__1cErs"
                        >
                          <div
                            class="Table-module_container__1CPuY table_hds-table-container__24SMd"
                            tabindex="0"
                          >
                            <table
                              class="Table-module_table__3pIAR table_hds-table__nft5d Table-module_dark__3WxCw Table-module_dense__NnaBV table_hds-table--dense__s0iM1 custom-theme-2"
                              data-testid="required-consents-group-0-table"
                              id="required-consents-group-0-table"
                            >
                              <thead>
                                <tr
                                  class="Table-module_headerRow__1P1Ko table_hds-table__header-row__2UlyP"
                                >
                                  <th
                                    scope="col"
                                  >
                                    Nimi
                                  </th>
                                  <th
                                    scope="col"
                                  >
                                    Evästeen asettaja
                                  </th>
                                  <th
                                    scope="col"
                                  >
                                    Käyttötarkoitus
                                  </th>
                                  <th
                                    scope="col"
                                  >
                                    Voimassaoloaika
                                  </th>
                                </tr>
                              </thead>
                              <tbody
                                class="Table-module_content__36xU2 table_hds-table__content__3WvvL"
                              >
                                <tr>
                                  <td
                                    data-testid="name-0"
                                  >
                                    wordpress_*, wp-settings-*
                                  </td>
                                  <td
                                    data-testid="hostName-0"
                                  >
                                    api.hel.fi
                                  </td>
                                  <td
                                    data-testid="description-0"
                                  >
                                    Wordpress-sisällönhallintajärjestelmän toimintaan liittyvä eväste.
                                  </td>
                                  <td
                                    data-testid="expiration-0"
                                  >
                                    Istunto-eväste tuhotaan, kun käyttäjä poistuu sivulta
                                  </td>
                                </tr>
                                <tr>
                                  <td
                                    data-testid="name-1"
                                  >
                                    linkedevents-api-prod-csrftoken
                                  </td>
                                  <td
                                    data-testid="hostName-1"
                                  >
                                    api.hel.fi
                                  </td>
                                  <td
                                    data-testid="description-1"
                                  >
                                    Eväste tapahtumien hakemisen valtuuttamiseen
                                  </td>
                                  <td
                                    data-testid="expiration-1"
                                  >
                                    1 vuosi
                                  </td>
                                </tr>
                                <tr>
                                  <td
                                    data-testid="name-2"
                                  >
                                    i18next
                                  </td>
                                  <td
                                    data-testid="hostName-2"
                                  >
                                    api.hel.fi
                                  </td>
                                  <td
                                    data-testid="description-2"
                                  >
                                    Eväste sivuston kielivalinnan muistamiseen
                                  </td>
                                  <td
                                    data-testid="expiration-2"
                                  >
                                    Istunto-eväste tuhotaan, kun käyttäjä poistuu sivulta
                                  </td>
                                </tr>
                              </tbody>
                            </table>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    class="CookieConsent-module_consentGroup__2i8Av"
                  >
                    <div
                      class="CookieConsent-module_titleWithCheckbox__1QVs0"
                    >
                      <div
                        class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz"
                        style="--label-font-size: var(--fontsize-heading-s);"
                      >
                        <input
                          aria-describedby="required-consents-group-1-description"
                          checked=""
                          class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
                          data-testid="required-consents-group-1-checkbox"
                          disabled=""
                          id="required-consents-group-1-checkbox"
                          name="required-consents-group-1-checkbox"
                          type="checkbox"
                          value=""
                        />
                        <label
                          class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
                          for="required-consents-group-1-checkbox"
                        >
                          Yhteiset evästeet
                        </label>
                      </div>
                    </div>
                    <div
                      class="CookieConsent-module_consentGroupContent__1x4Uq"
                    >
                      <p
                        aria-hidden="true"
                      >
                        Helsingin kaupungin palvelut käyttävät yhteisiä evästeitä. Tallennamme nämä suostumukset, jottei sinun tarvitse hyväksyä samoja evästeitä uudelleen kaupungin muissa palveluissa.
                      </p>
                      <div
                        id="required-consents-group-1-description"
                        style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
                      >
                        Helsingin kaupungin palvelut käyttävät yhteisiä evästeitä. Tallennamme nämä suostumukset, jottei sinun tarvitse hyväksyä samoja evästeitä uudelleen muissa kaupungin palveluissa.
                      </div>
                      <button
                        aria-expanded="false"
                        aria-label="Näytä yhteisiin evästesuostumuksiin liittyvien evästeiden tiedot"
                        class="CookieConsent-module_accordionButton__3R8Nn"
                        data-testid="required-consents-group-1-details-toggler"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          aria-label="angle-down"
                          class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                          role="img"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M12 13.5L17 8.5L18.5 10L12 16.5L5.5 10L7 8.5L12 13.5Z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </button>
                      <div
                        aria-label="Yhteiset evästeet"
                        class="Card-module_card__1WbKx custom-theme-3"
                        role="region"
                        style="display: none;"
                      >
                        <div
                          class="CookieConsent-module_dataTableContainer__1cErs"
                        >
                          <div
                            class="Table-module_container__1CPuY table_hds-table-container__24SMd"
                            tabindex="0"
                          >
                            <table
                              class="Table-module_table__3pIAR table_hds-table__nft5d Table-module_dark__3WxCw Table-module_dense__NnaBV table_hds-table--dense__s0iM1 custom-theme-4"
                              data-testid="required-consents-group-1-table"
                              id="required-consents-group-1-table"
                            >
                              <thead>
                                <tr
                                  class="Table-module_headerRow__1P1Ko table_hds-table__header-row__2UlyP"
                                >
                                  <th
                                    scope="col"
                                  >
                                    Nimi
                                  </th>
                                  <th
                                    scope="col"
                                  >
                                    Evästeen asettaja
                                  </th>
                                  <th
                                    scope="col"
                                  >
                                    Käyttötarkoitus
                                  </th>
                                  <th
                                    scope="col"
                                  >
                                    Voimassaoloaika
                                  </th>
                                </tr>
                              </thead>
                              <tbody
                                class="Table-module_content__36xU2 table_hds-table__content__3WvvL"
                              >
                                <tr>
                                  <td
                                    data-testid="name-0"
                                  >
                                    Evästesuostumukset
                                  </td>
                                  <td
                                    data-testid="hostName-0"
                                  >
                                    *.hel.fi
                                  </td>
                                  <td
                                    data-testid="description-0"
                                  >
                                    Sivusto käyttää tätä evästettä tietojen tallentamiseen siitä, ovatko kävijät antaneet hyväksyntänsä tai kieltäytyneet evästeiden käytöstä.
                                  </td>
                                  <td
                                    data-testid="expiration-0"
                                  >
                                    1 vuosi
                                  </td>
                                </tr>
                              </tbody>
                            </table>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </li>
              </ul>
            </div>
            <div
              class="CookieConsent-module_consentCategory__iJprz"
            >
              <div
                class="CookieConsent-module_titleWithCheckbox__1QVs0"
              >
                <div
                  class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz"
                  style="--label-font-size: var(--fontsize-heading-m);"
                >
                  <input
                    aria-describedby="optional-consents-description"
                    class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
                    data-testid="optional-consents-checkbox"
                    id="optional-consents-checkbox"
                    name="optional-consents-checkbox"
                    type="checkbox"
                    value=""
                  />
                  <label
                    class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
                    for="optional-consents-checkbox"
                  >
                    Valinnaiset evästeet
                  </label>
                </div>
              </div>
              <p
                aria-hidden="true"
              >
                Voit hyväksyä tai kieltää valinnaiset evästeet.
              </p>
              <div
                id="optional-consents-description"
                style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
              >
                Voit hyväksyä tai kieltää valinnaiset evästeet.
              </div>
              <ul
                class="CookieConsent-module_list__145_H"
              >
                <li>
                  <div
                    class="CookieConsent-module_consentGroup__2i8Av"
                  >
                    <div
                      class="CookieConsent-module_titleWithCheckbox__1QVs0"
                    >
                      <div
                        class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz"
                        style="--label-font-size: var(--fontsize-heading-s);"
                      >
                        <input
                          aria-describedby="optional-consents-group-0-description"
                          class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
                          data-testid="optional-consents-group-0-checkbox"
                          id="optional-consents-group-0-checkbox"
                          name="optional-consents-group-0-checkbox"
                          type="checkbox"
                          value=""
                        />
                        <label
                          class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
                          for="optional-consents-group-0-checkbox"
                        >
                          Matomo
                        </label>
                      </div>
                    </div>
                    <div
                      class="CookieConsent-module_consentGroupContent__1x4Uq"
                    >
                      <p
                        aria-hidden="true"
                      >
                        Matomo-analytiikkaohjelmistolla seurataan ja kehitetään verkkosivustoa käyttäjien tarpeita vastaavaksi.
                      </p>
                      <div
                        id="optional-consents-group-0-description"
                        style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
                      >
                        Matomo käyttää evästeitä.
                      </div>
                      <button
                        aria-expanded="false"
                        aria-label="Näytä Matomoon liittyvät evästetiedot."
                        class="CookieConsent-module_accordionButton__3R8Nn"
                        data-testid="optional-consents-group-0-details-toggler"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          aria-label="angle-down"
                          class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                          role="img"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M12 13.5L17 8.5L18.5 10L12 16.5L5.5 10L7 8.5L12 13.5Z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </button>
                      <div
                        aria-label="Matomo"
                        class="Card-module_card__1WbKx custom-theme-5"
                        role="region"
                        style="display: none;"
                      >
                        <div
                          class="CookieConsent-module_dataTableContainer__1cErs"
                        >
                          <div
                            class="Table-module_container__1CPuY table_hds-table-container__24SMd"
                            tabindex="0"
                          >
                            <table
                              class="Table-module_table__3pIAR table_hds-table__nft5d Table-module_dark__3WxCw Table-module_dense__NnaBV table_hds-table--dense__s0iM1 custom-theme-6"
                              data-testid="optional-consents-group-0-table"
                              id="optional-consents-group-0-table"
                            >
                              <thead>
                                <tr
                                  class="Table-module_headerRow__1P1Ko table_hds-table__header-row__2UlyP"
                                >
                                  <th
                                    scope="col"
                                  >
                                    Nimi
                                  </th>
                                  <th
                                    scope="col"
                                  >
                                    Evästeen asettaja
                                  </th>
                                  <th
                                    scope="col"
                                  >
                                    Käyttötarkoitus
                                  </th>
                                  <th
                                    scope="col"
                                  >
                                    Voimassaoloaika
                                  </th>
                                </tr>
                              </thead>
                              <tbody
                                class="Table-module_content__36xU2 table_hds-table__content__3WvvL"
                              >
                                <tr>
                                  <td
                                    data-testid="name-0"
                                  >
                                    _pk*
                                  </td>
                                  <td
                                    data-testid="hostName-0"
                                  >
                                    digia.fi
                                  </td>
                                  <td
                                    data-testid="description-0"
                                  >
                                    Matomo-tilastointijärjestelmän eväste
                                  </td>
                                  <td
                                    data-testid="expiration-0"
                                  >
                                    393 päivää
                                  </td>
                                </tr>
                              </tbody>
                            </table>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="CookieConsent-module_buttons__4uWoN"
          >
            <button
              class="Button-module_button__1msFE button_hds-button__2A0je Button-module_secondary__1nABp button_hds-button--secondary__1NOWS"
              data-testid="cookie-consent-approve-button"
              type="button"
            >
              <span
                class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
              >
                Hyväksy valitut evästeet
              </span>
            </button>
            <button
              class="Button-module_button__1msFE button_hds-button__2A0je Button-module_secondary__1nABp button_hds-button--secondary__1NOWS"
              data-testid="cookie-consent-approve-required-button"
              type="button"
            >
              <span
                class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
              >
                Hyväksy vain välttämättömät evästeet
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
